### ISSUE TITLE
Compiling Mono support with Visual Studio 2017 requires a change in SCons file

### ISSUE THREAD

**Issue by Firepal** [godotengine/godot#45839]
**Godot version:**
3.2

**OS/device including version:**
Windows 10

**Issue description:**
Currently, the SConstruct file contains [this code](https://github.com/godotengine/godot/blob/4211cb3dc7e812ad5c1a640134af8dc19e15c0dc/SConstruct#L387):
````python
    if not env.msvc:
        # Specifying GNU extensions support explicitly, which are supported by
        # both GCC and Clang. This mirrors GCC and Clang's current default
        # compile flags if no -std is specified.
        env.Prepend(CFLAGS=["-std=gnu11"])
        env.Prepend(CXXFLAGS=["-std=gnu++14"])
    else:
        # MSVC doesn't have clear C standard support, /std only covers C++.
        # We apply it to CCFLAGS (both C and C++ code) in case it impacts C features.
        env.Prepend(CCFLAGS=["/std:c++14"]) # <----- THIS !!!!
````

The issue is that some of the Mono bindings code like [modules\mono\mono_gd\gd_mono_marshal.cpp](https://github.com/godotengine/godot/blob/5f23fcf8479b1edaac540641cfdaf99aa5e0e226/modules/mono/mono_gd/gd_mono_marshal.cpp#L1336) uses the C++17 feature `[[maybe_unused]]` while MSVC2017 is told to compile at C++14 standard.

Thus MSVC2017 throws [C2429](https://docs.microsoft.com/en-us/cpp/error-messages/compiler-errors-1/compiler-error-c2429?view=msvc-150) ("The language feature requires a specific compiler option for support.")
when it sees a file that uses `[[maybe_unused]]`.

The fix I found to this is not too mean:
````diff
---        env.Prepend(CCFLAGS=["/std:c++14"])
+++        env.Prepend(CCFLAGS=["/std:c++17"])
````
but it's not certain whether or not this would break something else in the process. 
Maybe I should just throw a pull request and see what CI says?

**Comment by akien-mga** [comment]
The Mono code shouldn't use `[[maybe_unused]]` in the `3.2` branch, it should compile with C++14.

**Comment by NikodemL** [comment]
I also had to change it to `env.Prepend(CCFLAGS=["/std:c++17"])` to build the engine.
But it doesn't seem to break anything.

**Comment by Firepal** [comment]
@NikodemL Glad I'm not the only one, didn't hear a toot about this.

I was picturing a really old Visual Studio that doesn't have C++17 being unable to compile Godot,
but that's as stupid as trying to support IE 11 in these modern web times :P

**Comment by akien-mga** [comment]
Fixed by #46711.
